src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/DroolBase.java

```diff
@@ -129,8 +127,10 @@ RoleAssignment getRequestedCaseRole(RoleCategory roleCategory, String roleName,
             .classification(Classification.PUBLIC)
             .readOnly(true)
             .status(status)
-            .attributes(Map.of(attributeKey, convertValueJsonNode(attributeVal)))
+            .attributes(new HashMap<String, JsonNode>())
             .build();
+        ra.setAttribute(attributeKey, attributeVal);
```

Comment:
no need of ra.setAttribute(attributeKey, attributeVal);, it will take care by  attributes(Map.of(attributeKey, convertValueJsonNode(attributeVal)))

Reply:
The map.of* create an immutable map which was failing all the test cases, hence I need to replace that.

Reply:
you can create the map like new HashMap<>(Map.of(attributeKey, convertValueJsonNode(attributeVal))). I t will create the mutable map.